### DISCUSSION TITLE
How to install headless server on debian 8 ?

### DISCUSSION THREAD

**Discussion by uherbstsolace** [jamulussoftware/jamulus#1298] [Technical Support & Help]
Hi, I try to install the headless server on debian 8 (latest version... this is 3.7.0).

Deb installation does not work, because of missing QT5-dependencies.

Installing from source does not work:
```bash
....
src/main.cpp: In function ‘int main(int, char**)’:
src/main.cpp:110:19: error: ‘qInfo’ was not declared in this scope
             qInfo() << "- server mode chosen";
                   ^
src/main.cpp:123:19: error: ‘qInfo’ was not declared in this scope
             qInfo() << "- no GUI mode chosen";
                   ^
src/main.cpp:137:19: error: ‘qInfo’ was not declared in this scope
             qInfo() << "- licence required";
                   ^
....
```

Any ideas how to get a headless server running on debian 8 ? 
(and dont propose to upgrade debian... I know, this version is old, but I can't change that)

**Comment by hoffie** [comment]
Hi,

Some options I can think of:
- Attempt to patch out everything which requires a newer Qt version (e.g. try replacing with `qInfo()`, `qDebug()`, etc. by `std::cout`). However, I fear that there are other places which may break and may not be that simple to work around.
- Use a chroot or container based on a newer Debian version. Not sure if/how well containers work on Debian 8 though. Chroots should work though, you could probably use `debootstrap`. Besides requiring some space, this sounds like the best option to me.
- Get a newer Qt from somewhere and install it to an alternative location (unpacking -- not installing! -- newer packages from Debian or ubuntu may work; should be unpacked to some designed directory). This would require fiddling with LD_* variables though.
- Attempt to create a statically linked Jamulus build on a newer machine and try to run the resulting binary.

I'm trying to be helpful, but can't resist saying that according to the end-of-support in 2018, Debian 8 should probably not be run in any untrusted, networked environment.

Kind regards,
Christian
